include/net/lwm2m.h

```diff
@@ -1062,6 +1062,22 @@ int lwm2m_engine_update_service_period(k_work_handler_t service, uint32_t period
  */
 int lwm2m_update_device_service_period(uint32_t period_ms);
 
+/**
+ * @brief Check whether a path is observed
+ *
+ * @param[in] pathstr LwM2M path string to check, e.g. "3/0/1"
+ *                    Note: level 4 (resource-instance) is treated the exact same
+ *                    as level 3 (resource) since there exist no resource instance
+ *                    specific observations in LwM2M.
+ *
+ * @return 1 when there exists an observation of the same level
+ *         or lower as the given path, 0 otherwise. Negative if the given string
+ *         is not a valid LwM2M-path.
+ *         E.g. true if path refers to a resource and the parent object has an
+ *         observation, false for the inverse.
+ */
+int lwm2m_engine_path_is_observed(char *pathstr);
```

Comment:
Referring to #42565, maybe make the parameter `const` rightaway?

Reply:
not possible without casts right now before #42565 is merged, because string_to_path expects a non-const argument

Reply:
Ok, how about adding a DNM for now and wait for https://github.com/zephyrproject-rtos/zephyr/pull/42565 to get merged first (note however that this might take some time as we're currently in a code freeze).

Reply:
sounds good to me.

Reply:
rebased and adjusted now that #42565 got merged.